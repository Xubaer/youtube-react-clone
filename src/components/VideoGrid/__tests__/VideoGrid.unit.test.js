import React from 'react';
import { shallow } from 'enzyme';
import { VideoGrid } from '../VideoGrid';

describe('VideoGrid', () => {
    test('renders without props', () => {
        const wrapper = shallow(<VideoGrid/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders with empty props', () => {
        const wrapper = shallow(<VideoGrid title=''/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders with title', () => {
        const wrapper = shallow(<VideoGrid title='Trending'/>);
        expect(wrapper).toMatchSnapshot();
    });

})