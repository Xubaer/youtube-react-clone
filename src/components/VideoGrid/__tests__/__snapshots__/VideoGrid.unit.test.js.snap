// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders with empty props 1`] = `
<Fragment>
  <VideoGridHeader
    title="trending"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders with title 1`] = `
<Fragment>
  <VideoGridHeader
    title="trending"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders without props 1`] = `
<Fragment>
  <VideoGridHeader
    title="trending"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;
